File: clients/java-client/src/main/java/io/ehdev/conrad/client/java/VersionServiceConfiguration.java
package io.ehdev.conrad.client.java;

import java.util.concurrent.TimeUnit;

public class VersionServiceConfiguration {
    String repoId;
    String providerBaseUrl;
    int timeout = (int)TimeUnit.MILLISECONDS.convert(10, TimeUnit.SECONDS);

    public String getRepoId() {
        return repoId;
    }

    public void setRepoId(String repoId) {
        this.repoId = repoId;
    }

    public String getProviderBaseUrl() {
        return providerBaseUrl;
    }

    public void setProviderBaseUrl(String providerBaseUrl) {
        this.providerBaseUrl = providerBaseUrl;
    }

    public int getTimeout() {
        return timeout;
    }

    public void setTimeout(int timout) {
        this.timeout = timout;
    }
}

File: clients/java-client/src/main/kotlin/io/ehdev/conrad/client/java/VersionServiceConfiguration.kt
package io.ehdev.conrad.client.java

import okhttp3.OkHttpClient
import java.util.concurrent.TimeUnit

open class VersionServiceConfiguration(val client: OkHttpClient) {
    var repoName: String? = null
    var projectName: String? = null
    var applicationUrl: String? = null
    var authToken: String? = null

    constructor() : this(OkHttpClient.Builder().connectTimeout(10, TimeUnit.SECONDS).readTimeout(10, TimeUnit.SECONDS).build()) { }
}
